rllib/algorithms/impala/impala_learner.py

```diff
@@ -166,9 +225,8 @@ def step(self):
                 ma_batch_on_gpu = self._in_queue.sample()
             else:
                 # Queue is empty: Sleep a tiny bit to avoid CPU-thrashing.
```

Comment:
Usually the scheduler makes a good job in providing runtime to each process. This while loop could slow things down and in worst case block.

Reply:
I do think we have to sleep here (or at least be careful with removing this logic).

The "queue" here is a `deque`, which doesn't have the GIL-release logic of a `Queue.get()`. Looping here with `while True:` without sleeping would certainly harm performance.